apps/editing-toolkit/editing-toolkit-plugin/block-patterns/class-block-patterns-from-api.php

```diff
@@ -72,75 +67,90 @@ private function register_patterns() {
 			}
 		}
 
-		$pattern_categories = array();
-		$block_patterns     = $this->get_patterns();
-
-		foreach ( (array) $block_patterns as $pattern ) {
-			foreach ( (array) $pattern['categories'] as $slug => $category ) {
-				$pattern_categories[ $slug ] = array( 'label' => $category['title'] );
-			}
-		}
+		// For every pattern source site, fetch the patterns
+		foreach( $this->patterns_sources as $patterns_source ) {
```

Comment:
Sanity check: 

@andrewserong 😄 

This works, just wanted your opinion on whether it'd make more sense to make 2 x calls to the patterns lib on the backend. 

My other approach was going to be modifying the query params, e.g., 

`'patterns_source'  => '(string) Comma-separated list of slugs to identify a default patterns source site. Default: `dotcompatterns.wordpress.com`',`

And then return the combined array of sources. The advantage I see in that way is that, aside from being only 1 x API call, is we could do any required sorting/filtering before we send the response (assuming we need to).

Do you envisage any caching issues?

Reply:
At this stage, I think of the two options, I'd lean toward doing 2 x API calls.

When making a change to the `ptk/patterns` endpoint, it's good for us to think about how we'd like to call that endpoint in the long-term. At the moment, having the two separate source sites for FSE and standard patterns sounds like a short-term solution, before ultimately consolidating the patterns in the one source site. So doing the workaround in the ETK here in this PR makes good sense to me (firing off two API calls), but let's see if it impacts initial page load times.

Longer-term I can see a good case for getting the API endpoint to support pulling from multiple source sites at once, and as you say, combining / sorting or filtering the results before returning a response. Something like that could be useful for pulling content from across all of the starter source sites, too. But there's a few different ways we could implement that, which is a bit beyond the scope of what we need to do to get this working. So I'd think of that as a good area to explore for if and when we have more use cases for it.

So for now, and since we're working to support testing out the FSE experience, I think I'd go for the shorter-term solution of the multiple API calls, to avoid adding extra complexity to the endpoint. If it turns out it's a noticeable performance hit, then I think that'd easily justify the extra work exploring adding support for multiple source sites to the endpoint itself.

What do you think? I'm definitely happy for us to explore further API changes if you think we'll need it!

Reply:
The change to the `$patterns_cache_key` looks good to me, too 👍

Reply:
Thanks for the confirmation. I wasn't sure, but now I feel better about it 🙇